front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -282,14 +282,14 @@ export function updateDaySinceDeparture(
 }
 
 export function durationSinceStartTime(
-  startTime?: IsoDateTimeString,
+  startTime?: Date,
   stepTimeDays?: TimeExtraDays
 ): IsoDurationString | null {
   if (!startTime || !stepTimeDays?.time || stepTimeDays?.daySinceDeparture === undefined) {
     return null;
   }
   const start = dayjs(startTime);
-  const step = dayjs(`${startTime.split('T')[0]}T${stepTimeDays.time}`).add(
+  const step = dayjs(`${start.format('YYYY-MM-DD')}T${stepTimeDays.time}`).add(
```

Comment:
Note: we should really get rid of this kind of string manipulation…

Reply:
can’t completely avoid it, but we shouldn’t mix very low level manipulation with higher abstraction

Reply:
I mean I’m not overly shocked to have string manipulation inside low level utils function. As long as we can hide those details and are able to think about the higher level code/business code without all these low levels details cluttering the code.

But yeah for dates we have decents apis, we might as well use them here